include/nuttx/fs/fs.h

```diff
@@ -209,13 +211,19 @@ struct file_operations
   int     (*close)(FAR struct file *filep);
   ssize_t (*read)(FAR struct file *filep, FAR char *buffer, size_t buflen);
   ssize_t (*write)(FAR struct file *filep, FAR const char *buffer,
-                   size_t buflen);
+                    size_t buflen);
   off_t   (*seek)(FAR struct file *filep, off_t offset, int whence);
   int     (*ioctl)(FAR struct file *filep, int cmd, unsigned long arg);
+  int     (*truncate)(FAR struct file *filep, off_t length);
+  int     (*mmap)(FAR struct file *filep, FAR void **addr, off_t start,
+                  size_t length);
+  int     (*munmap)(FAR struct task_group_s *group, FAR struct mm_map_s *map,
```

Comment:
why need struct task_group_s

Reply:
This is to handle the special case of process exit. Normally munmap needs to remove the mapping and also in case of CONFIG_BUILD_KERNEL also do the real MMU unmapping. In case of process exit the MMU unmapping can't be done (since there may be some other address environment currently in use), and also removal from the group can be done outside.

Reply:
This special case can indicate by setting mm_map_s->group to NULL?

Reply:
I don't like storing the group into mm_map_s. Since the mm_map_s is stored inside the group, there is no point in wasting memory by storing it; it has to be the same :)

Just indicating somehow that the process is about the exit and the current address evironment should not be touched is enough for the munmap call. In this case the driver needs to do all the cleanup (kmm_free etc.), but not touch group specific things like real MMU mappings.

A simple way to do this is to just either pass the current "group" or NULL to the unmap.

Reply:
Ok.